sentry-core/src/main/java/io/sentry/core/util/VisibleForTesting.java

```diff
@@ -0,0 +1,20 @@
+package io.sentry.core.util;
+
+import static java.lang.annotation.ElementType.CONSTRUCTOR;
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.TYPE;
+import static java.lang.annotation.RetentionPolicy.CLASS;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+/**
+ * Annotates a declaration as visible solely for testing purposes. Such declaration should not be
+ * used in production code.
+ */
+@Documented
+@Target({TYPE, METHOD, FIELD, CONSTRUCTOR})
+@Retention(CLASS)
```

Comment:
should it not be `SOURCE` ?

Reply:
Depends :) If we want some static analysis that is able to work on the bytecode level then we need `CLASS` so that the annotation is retained in the class file. `SOURCE` would make it only available during compilation and not put into the compiled jarfile. E.g. if we were linting `sentry-core` only as a dependency, as we are atm, then we have to rely on the bytecode analysis.

Reply:
mmm, that's interesting, but I'm not an expert about this matter, just did a quick research and it seems that `android lint` does source code analyses and `findbugs` does bytecode analyses.
could you just double-check it? I'm not sure how it works when a java lib. is a dependency for an Android lib.
if it should be CLASS, all good :D

Reply:
AFAIK Lint is not applied on bytecode but on an AST derived from source code files.
Guava as well has got a `VisibleForTesting` annotation which has `CLASS` retention.

However, even if our annotation is set to have retention `CLASS` I don't think the current available Lint check would take it into account (unless there is a way to configure it) nor any static analysis tool will be able to process it by default.

So now I am wondering: what is the advantage of adding such annotations in our package scope?

Reply:
I think you're right - lint seems to only be using the sources of java libs available during the build. I tried depending in `android-core` on a test lib I created which uses `Objects.hash()` in a function called from `android-core`. The lib is built separately and so lint doesn't have access to its sources. And our build passes even though we're clearly using a lib that can break at runtime due to API incompat. . 

So in another words lint cannot check that e.g. some random java lib from maven central doesn't use an API that is not present in the required API level. That is kinda disappointing and limits what `checkDependencies` can be used for. I learned something today :)

Reply:
> So now I am wondering: what is the advantage of adding such annotations in our package scope?

Given the findings above, I don't think it makes sense to have that annotation or indeed just have it at `SOURCE` level so that it doesn't "pollute" the compiled artifacts.

Reply:
Yeah I would keep for internal reference and set it to `SOURCE` retention.

Reply:
> So in another words lint cannot check that e.g. some random java lib from maven central doesn't use an API that is not present in the required API level. That is kinda disappointing and limits what `checkDependencies` can be used for. I learned something today :)

Absolutely, `checkDependencies` is basically for dependent library projects (in Gradle terminology)